settings.gradle.kts

```
@@ -81,13 +81,13 @@ file("services").listFiles().forEach {
 val compositeProjectList = try {
     val localProperties = java.util.Properties()
     localProperties.load(File(rootProject.projectDir, "local.properties").inputStream())
-    val filePaths = localProperties.getProperty("compositeProjects")
-        ?.splitToSequence(",") // Split comma delimited string into sequence
-        ?.map { it.replaceFirst("^~".toRegex(), System.getProperty("user.home")) } // expand user dir
-        ?.filter { it.isNotBlank() }
-        ?.map { file(it) } // Create file from path
-        ?.toList()
-        ?: emptyList()
+    val propertyVal = localProperties.getProperty("compositeProjects") ?: "../smithy-kotlin"
```

Would it be better to throw an exception if there is no `compositeProjects` property (such as with `checkNotNull { ... }`) so it can fallback to the `catch (e: Throwable)` logic on L96? It would remove the (small) duplication

---

`local.properties` is heavily used for Android development so it's entirely reasonable that the file may not have this key.

---

if it's missing, that's fine, it will still default to `../smithy-kotlin`, just in a different way

---

Whoops, shipped as soon as I saw approvals but before I saw comments. Yes, it would've probably been better to clean up the duplication.